src/OpenIDConnect.jl

```diff
@@ -246,7 +240,9 @@ function flow_validate_id_token(ctx::OIDCCtx, jwt::JWT)
         if isvalid
             validator = ctx.validator
             if (time() - ctx.last_key_refresh) >= ctx.key_refresh_secs
-                refresh!(validator)
+                jstr = String(HTTP.get(ctx.validator.url; ctx.http_tls_opts...).body)
+                keys = JSON.parse(jstr)["keys"]
+                refresh!(keys, Dict{String,JWK}();)
```

Comment:
I think the refreshed keys are not being set in the validator here. Did you mean this to be something like the below?
```julia
keysetdict = Dict{String,JWK}()
refresh!(keys, keysetdict)
validator.keys = keysetdict
```

Reply:
You're absolutely right